packages/bot-web-ui/src/pages/bot-builder/quick-strategy/analytics/rudderstack-quick-strategy.ts

```diff
@@ -0,0 +1,79 @@
+import { Analytics, TEvents } from '@deriv-com/analytics';
+import { ACTION, form_name, type TFormStrategy, type TSelectedStrategy } from './constants';
+import { getRsStrategyType, getTradeParameterData } from './utils';
+
+export const rudderStackSendQsOpenEventFromDashboard = () => {
+    Analytics.trackEvent('ce_bot_form', {
+        action: ACTION.OPEN,
+        form_name,
+        subpage_name: 'bot_builder',
+        subform_name: 'quick_strategy',
+        subform_source: 'dashboard',
+    });
+};
+
+export const rudderStackSendQsOpenEventFromBotBuilder = () => {
+    Analytics.trackEvent('ce_bot_form', {
+        action: ACTION.OPEN,
+        form_name,
+        subpage_name: 'bot_builder',
+        subform_name: 'quick_strategy',
+        subform_source: 'bot_builder',
+    });
+};
+
+export const rudderStackSendQsCloseEvent = ({
+    quick_strategy_tab,
+    selected_strategy,
+}: TEvents['ce_bot_form'] & TFormStrategy) => {
+    Analytics.trackEvent('ce_bot_form', {
+        action: ACTION.CLOSE,
+        form_name,
+        subpage_name: 'bot_builder',
+        subform_name: 'quick_strategy',
+        quick_strategy_tab,
+        strategy_name: getRsStrategyType(selected_strategy),
+    });
+};
+
+export const rudderStackSendQsRunStrategyEvent = ({
+    form_values,
+    selected_strategy,
+}: TEvents['ce_bot_form'] & TFormStrategy) => {
+    Analytics.trackEvent('ce_bot_form', {
+        action: ACTION.RUN_QUICK_STRATEGY,
+        form_name,
+        subform_name: 'quick_strategy',
+        ...getTradeParameterData({ form_values, selected_strategy }),
+    });
+};
+
+export const rudderStackSendQsEditStrategyEvent = ({
+    form_values,
+    selected_strategy,
+}: TEvents['ce_bot_form'] & TFormStrategy) => {
+    Analytics.trackEvent('ce_bot_form', {
+        action: ACTION.EDIT_QUICK_STRATEGY,
+        form_name,
+        subform_name: 'quick_strategy',
+        ...getTradeParameterData({ form_values, selected_strategy }),
```

Comment:
@farabi-deriv bhai the output from this function could be nested and the nested object is not supported currently in our Rudderstack. Can you please check this?

Reply:
@shafin-deriv sure bhai. Done ✅